projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -103,8 +103,21 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
     @Input()
     canOpen = true;
 
+    private readonly onHoverTarget$ = new Subject<HTMLElement>();
+
+    private readonly _hover$ = combineLatest([
+        this.onHoverTarget$,
+        this.hover$ || EMPTY,
+    ]).pipe(
+        map(
+            ([target, hovered]: [HTMLElement, boolean]) =>
+                hovered && this.computedHost.contains(target),
+        ),
+        share(),
```

Comment:
Do we need a `share` here? Seems like it is only used in one place. Maybe we should add `share` to `open$` instead?